homeassistant/components/melcloud/config_flow.py

```diff
@@ -126,3 +133,67 @@ async def async_step_import(self, user_input):
         if result["type"] == FlowResultType.CREATE_ENTRY:
             await async_create_import_issue(self.hass, self.context["source"], "", True)
         return result
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle initiation of re-authentication with MELCloud."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with MELCloud."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None and self.entry:
+            aquired_token, errors = await self.async_reauthenticate_client(user_input)
+
+            if not errors:
+                self.hass.config_entries.async_update_entry(
+                    self.entry,
+                    data={CONF_TOKEN: aquired_token},
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self.entry.entry_id)
+                )
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+            ),
+            errors=errors,
+        )
+
+    async def async_reauthenticate_client(
+        self, user_input: dict[str, Any]
+    ) -> tuple[str | None, dict[str, str]]:
+        """Reauthenticate with MELCloud."""
+        errors: dict[str, str] = {}
+        acquired_token = None
+
+        try:
+            async with asyncio.timeout(10):
+                acquired_token = await pymelcloud.login(
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                    async_get_clientsession(self.hass),
+                )
+        except (ClientResponseError, AttributeError) as err:
```

Comment:
Why do we catch `AttributeError`? That should normally never be needed if the design is correct.